python/tvm/relax/expr.py

```diff
@@ -384,10 +384,6 @@ class Constant(ExprWithOp):
     def __init__(
         self, data: tvm.nd.NDArray, struct_info: Optional[StructInfo] = None, span: Span = None
     ) -> None:
-        if struct_info is not None:
-            struct_info = tvm.runtime.convert_to_object(struct_info)
-            if not isinstance(struct_info, StructInfo):
```

Comment:
is this unintentional?

Reply:
No. This is intentional. To support R.dist.const, we need to pass in the device mesh and placement, so I add a parameter `struct_info` into R.dist.const. For R.const, the struct info is inferred from data and dtype, while for R.dist.const, we directly use the annotated struct_info

Reply:
Can we do it differently without changing the base IR? One way i can think of is to enable R.distribute right after constant, so the normal Tensor are treated as non-replicated tensor on host